### PR TITLE
[FLINK-6543] [table] Deprecate toDataStream

### PR DESCRIPTION
**Pull request by twalthr**
This deprecates toDataStream in Scala and Java Table API.

### CODE REVIEW

**Comment by twalthr** [comment]
I forgot to update the documentation. Will do this later.

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/java/StreamTableEnvironment.scala [hunk 1]
```diff
@@ -144,13 +144,119 @@ class StreamTableEnvironment(
...
+    * If add and retract messages are required, use [[toRetractStream()]].
+    *
+    * @param table The [[Table]] to convert.
+    * @param typeInfo The [[TypeInformation]] that specifies the type of the [[DataStream]].
+    * @tparam T The type of the resulting [[DataStream]].
+    * @return The converted [[DataStream]].
+    * @deprecated This method only supports conversion of append-only tables. In order to
+    *            make this more explicit in the future, please use toAppendStream() instead.
+    */
+  def toDataStream[T](table: Table, typeInfo: TypeInformation[T]): DataStream[T] =
```
Comment (fhueske) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/java/StreamTableEnvironment.scala]: add `@Deprecated` annotation

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/java/StreamTableEnvironment.scala [hunk 2]
```diff
@@ -144,13 +144,119 @@ class StreamTableEnvironment(
...
+    *
+    * @param table The [[Table]] to convert.
+    * @param clazz The class of the type of the resulting [[DataStream]].
+    * @param queryConfig The configuration of the query to generate.
+    * @tparam T The type of the resulting [[DataStream]].
+    * @return The converted [[DataStream]].
+    * @deprecated This method only supports conversion of append-only tables. In order to
+    *            make this more explicit in the future, please use toAppendStream() instead.
+    */
+  def toDataStream[T](
```
Comment (fhueske) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/java/StreamTableEnvironment.scala]: add `@Deprecated` annotation

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/java/StreamTableEnvironment.scala [hunk 3]
```diff
@@ -144,13 +144,119 @@ class StreamTableEnvironment(
...
+    *
+    * @param table The [[Table]] to convert.
+    * @param typeInfo The [[TypeInformation]] that specifies the type of the [[DataStream]].
+    * @param queryConfig The configuration of the query to generate.
+    * @tparam T The type of the resulting [[DataStream]].
+    * @return The converted [[DataStream]].
+    * @deprecated This method only supports conversion of append-only tables. In order to
+    *            make this more explicit in the future, please use toAppendStream() instead.
+    */
+  def toDataStream[T](
```
Comment (fhueske) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/java/StreamTableEnvironment.scala]: add `@Deprecated` annotation

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/java/StreamTableEnvironment.scala [hunk 4]
```diff
@@ -144,13 +144,119 @@ class StreamTableEnvironment(
...
+    *
+    * @param table The [[Table]] to convert.
+    * @param clazz The class of the type of the resulting [[DataStream]].
+    * @tparam T The type of the resulting [[DataStream]].
+    * @return The converted [[DataStream]].
+    * @deprecated This method only supports conversion of append-only tables. In order to
+    *            make this more explicit in the future, please use toAppendStream() instead.
+    */
+  @Deprecated
+  def toDataStream[T](table: Table, clazz: Class[T]): DataStream[T] = toAppendStream(table, clazz)
```
Comment (fhueske) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/java/StreamTableEnvironment.scala]: This method is still referenced from the `java` `package-info.java` file and the `java` `SqlITCase`.

**Comment by fhueske** [comment]
Will fix the remaining issues and merge this for RC2 of Flink 1.3.0.

**Pull request closed without merging**